Build libario shared so that we can link the plugins against it.

--- origsrc/ario-1.3/Makefile.am	2008-12-09 12:48:57.000000000 -0600
+++ src/ario-1.3/Makefile.am	2009-07-23 01:45:19.366048300 -0500
@@ -6,6 +6,6 @@ else
         SUBDIRS =
 endif
 SUBDIRS += data\
+           src\
            plugins\
-           po\
-           src
+           po
--- origsrc/ario-1.3/configure.ac	2009-05-21 13:06:33.000000000 -0500
+++ src/ario-1.3/configure.ac	2009-07-23 01:33:55.383097200 -0500
@@ -252,6 +252,9 @@ AC_SUBST(PLUGINDIR)
 PLUGIN_DATA_DIR='${pkgdatadir}/plugins'
 AC_SUBST(PLUGIN_DATA_DIR)
 
+PLUGIN_LIBTOOL_FLAGS='-module -avoid-version -no-undefined $(top_builddir)/src/libario.la'
+AC_SUBST(PLUGIN_LIBTOOL_FLAGS)
+
 AC_OUTPUT([
 Makefile
 bindings/Makefile
--- origsrc/ario-1.3/src/Makefile.am	2009-05-12 13:02:37.000000000 -0500
+++ src/ario-1.3/src/Makefile.am	2009-07-23 02:02:37.137193600 -0500
@@ -1,7 +1,7 @@
 BUILT_SOURCES = 
 bin_PROGRAMS = ario
 
-noinst_LTLIBRARIES = libario.la
+lib_LTLIBRARIES = libario.la
 
 INCLUDES = -Wall -Wimplicit -Wunused-label \
 	   -Wmissing-declarations -Wmissing-prototypes \
@@ -16,6 +16,8 @@ INCLUDES = -Wall -Wimplicit -Wunused-lab
 ario_SOURCES = ario-main.c
 ario_LDADD = libario.la $(DEPS_LIBS)
 
+libario_la_LDFLAGS = -avoid-version -no-undefined
+
 libario_la_LIBADD = \
         $(DEPS_LIBS)\
         $(AVAHI_LIBS)\
@@ -183,7 +185,7 @@ if USE_XMMS2
 endif
 
 if ENABLE_PYTHON
-        ario_SOURCES += plugins/ario-python-module.c\
+        libario_la_SOURCES += plugins/ario-python-module.c\
                         plugins/ario-python-module.h\
                         plugins/ario-python-plugin.c\
                         plugins/ario-python-plugin.h
